### PR TITLE
LPS-171926 Remove Stream usages in portal-vulcan-api

### PR DESCRIPTION
**Pull request by julschong**

### CODE REVIEW

**Comment by zhangyan0701** [comment]
ci:test:headless-frontend-functional

**Comment by zhangyan0701** [comment]
ci:test:headless

**Comment by catarineaguiar** [comment]
no regressions found.

**Comment by julschong** [comment]
pushed to removed labeled for loop

**Comment by julschong** [comment]
ci:test:sf

**Comment by julschong** [comment]
ci:test:relevant

**Comment by zhangyan0701** [comment]
ci:test:headless

**Comment by zhangyan0701** [comment]
No regressions found.

File: modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/util/OpenAPIUtil.java [hunk 1]
```diff
@@ -167,34 +169,46 @@ private static boolean _hasOKResponseContentSchemaReferenceLike(
...
+						Schema::getReference
+					).map(
+						reference -> StringUtil.equals(
+							name,
+							reference.substring(reference.lastIndexOf('/') + 1))
+					).orElse(
+						false
+					)) {
+
+					hasOKResponse = true;
```
Comment (dantewang) [inline modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/util/OpenAPIUtil.java:200]: @julschong I think you can return true here, and return false at line 211, so you don't need a boolean flag.

BTW this pull will only deal with `Stream<>` usages?

**Review by dantewang — CHANGES_REQUESTED** [review]

File: modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/util/OpenAPIUtil.java (hunk 1 above)
Comment (julschong) [reply modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/util/OpenAPIUtil.java:200]: Hi Dante,
I'll take a look at just returning true
yes on this pull only doing Stream. If you would like, I can add on Optional

Comment (julschong) [reply modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/util/OpenAPIUtil.java:200]: @dantewang you're absolutely right about returning true, I wanted to do Optional separately but I think all in one pull may be better organization in general. I will add Optional on

**Pull request closed without merging**